internal/handler/job/events/events.go

```diff
@@ -10,16 +10,54 @@ package events
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	"github.com/go-openapi/strfmt"
 	"github.com/google/uuid"
-	"github.com/olebedev/emitter"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 	"github.com/siemens/wfx/generated/api"
 	"github.com/siemens/wfx/middleware/logging"
 )
 
+type Subscriber struct {
+	Events  <-chan JobEvent // channel which receives the events for the subscriber
+	Backlog *Backlog
+
+	id            string        // unique identifier
+	ch            chan JobEvent // internal RW of Events
+	lastBlock     time.Time     // time of the last failed attempt to send a message
+	graceDuration time.Duration // duration after which non-responsive subscribers are dropped
+
+	jobIDSet    map[string]any // job filter
+	clientIDSet map[string]any // clientID filter
+	workflowSet map[string]any // workflow filter
+	tags        []string       // tags to apply
+}
+
+type Backlog struct {
+	data []JobEvent
+	mu   sync.Mutex
+}
+
+func (b *Backlog) Push(event JobEvent) {
```

Comment:
Hm, `Push()` and `Pop()` are suggestive for a stack structure, while this is, however, a queue where you put stuff in at the end and remove from the beginning, i.e., head. Probably `Enq()` and `Deq()` are better terms?